core/base/v7/inc/ROOT/TDirectory.h

```diff
@@ -96,47 +92,51 @@ class TDirectory {
   /// \param name  - Key of the object.
   /// \param args  - arguments to be passed to the constructor of `T`
   template <class T, class... ARGS>
-  std::weak_ptr<T> Create(const std::string& name, ARGS... args) {
-    auto ptr = std::make_shared<ToContentType_t<T>>(std::forward<ARGS...>(args...));
+  decltype(auto) Create(const std::string& name, ARGS&&... args) {
```

Comment:
@karies at first, a general question: what is the preferred way to update PR: amend-commiting + forced pushing or just adding new commits?

About `decltype(auto)`: which pointer should I return: `weak_ptr` or `shared_ptr`? I believe `shared_ptr`, because I don't see reasons why user might want not to hold ownership of object (s)he creates.